fluence-js/src/Session.ts

```diff
@@ -152,19 +148,104 @@ export class Session {
             this.markSessionAsClosed(this.closedStatus)
         }
 
-        detailedDebug("start request");
-
         // increments counter at the start, if some error occurred, other requests will be canceled in `cancelAllPromises`
         let currentCounter = counter ? counter : this.getCounterAndIncrement();
 
         let signed = withSignature(payload, currentCounter, privateKey);
-        let tx = `${this.session}/${currentCounter}\n${signed}`;
+        let path = this.targetKey(currentCounter);
+        let tx = `${path}\n${signed}`;
+
+        return  {
+            path: path,
+            payload: tx
+        }
+    }
+
+    async query(path: string): Promise<RequestState<Option<Result>>> {
+        detailedDebug("start query");
+
+        // throws an error immediately if the session is closed
+        if (this.closed) {
+            return {
+                status: RequestStatus.E_SESSION_CLOSED,
+                error: error(`The session was closed. Cause: ${this.closedStatus}`)
+            };
+        }
+
+        if (this.closing) {
+            this.markSessionAsClosed(this.closedStatus)
+        }
+
+        let queryResult;
```

Comment:
Could be simplified like this:
```suggestion
        try {
            return {
                status: RequestStatus.OK,
                result: await this.tm.abciQuery(path)
            };
        } catch (err) {
            return {
                status: RequestStatus.E_REQUEST,
                error: error(`Request error on query occured. Request path: ${path}, error: ${JSON.stringify(err)}`),
            }
        }
```